scapy/utils.py

```diff
@@ -488,17 +496,24 @@ def do_graph(graph,prog=None,format=None,target=None,type=None,string=None,optio
     start_viewer=False
     if target is None:
         if WINDOWS:
-            tempfile = get_temp_file() + "." + format
+            tempfile = get_temp_file(autoext=format)
             target = "> %s" % tempfile
             start_viewer = True
         else:
             target = "| %s" % conf.prog.display
+    elif os.path.exists(os.path.split(os.path.abspath(target))[0]):
```

Comment:
OK I think I was wrong and maybe we want to really change this API. We could allow for user either a string (=> a filename), or an opened file descriptor. Thanks to this, we could use `stdout=subprocess.PIPE` to redirect the data, and this should work with Windows. What do you think?

Reply:
Seems a good idea. Will do

Reply:
Even better, i'm using `stdout` directly as a redirection (it can accepts a file descriptor/file object), to handle all cases